### PR TITLE
Add support for REJECT instead of DROP as deny action

### PR DESCRIPTION
**Pull request by olljanat**
## Description
New feature, allowing you REJECT denied traffic instead of DROP it.

This is useful in cases where you want to notify callers that the
path is disallowed instead of just letting their requests time out.

In tests:
- Using DROP action on iptables mode
- Using REJECT action on IPVS mode

## Related issues/PRs
* #4500
* #5256

## Todos

- [x] Tests
- [x] Documentation
- [x] Release note

## Release Note

```release-note
Add ability to set the deny action as REJECT instead of DROP
```

### CODE REVIEW

**Comment by olljanat** [comment]
Now it have at least some set of tests but someone how is more familiar with this codebase most probably should look about those and tell if they are good enough? Felix local tests with `make test` at least pass now (except one but that looks more like issue on my dev environment).

However would be interesting to see what is result of those E2E tests as last time those failed to check-fmt (which is now fixed) on single test code file.

**Comment by olljanat** [comment]
@caseydavenport Oh, there was kind of stupid mistake that I didn't run all those code generation scripts and e2e stopped before it really started. I managed to fix that by enabling Semaphore CI to my fork it looks to be quite tricky to get all tests working in there. So please trigger e2e tests again in here.

**Comment by olljanat** [comment]
@lwr20 thank you. Calicoctl failure was valid as I had forgot `gofmt` from there (it is fixed now).

However it it very hard to see that how that "UT/FV tests on new kernel" test failure
```
Summarizing 1 Failure:
[Fail] _BPF_ _BPF-SAFE_ BPF tests (tcp, ct=false, log=debug, tunnel=ipip, dsr=true) (kubernetes backend) with BPF disabled to begin with [It] should keep a connection up between hosts and remote workloads when BPF is enabled
```
would have anything to do with this change. Has it it been flaky earlier? Full log can be found from [here](https://tigera.semaphoreci.com/jobs/e2c2365e-4b1e-4dc1-ab21-9a66d3d20129)

**Comment by olljanat** [comment]
@caseydavenport any idea if "UT/FV tests on new kernel" failure can be caused by my changes or if that was just CI hiccup?

**Comment by olljanat** [comment]
"UT/FV tests on new kernel" definitely looks flaky because it failed also on #6899 which only updates documentation.

Anyway, rebased and fixed conflicts with #6862 so new run for CI would be nice.

**Comment by olljanat** [comment]
@lwr20 thanks. now "UT/FV tests on new kernel" passed so it was CI hiccup earlier. Looks that I did bad rebase last time for calicoctl (it is a bit confusing that `make gen-crds` is not enough but `gofmt -w` is needed after that). Fixed that one and rebased to fix conflicts with #6912

So I assume that this will pass CI now and it would be nice to get proper code review for this PR.

**Comment by olljanat** [comment]
ping @lwr20 @caseydavenport Code review please

**Comment by olljanat** [comment]
Rebased, still waiting for code review.

File: felix/rules/static_test.go [hunk 1]
```diff
@@ -78,6 +78,13 @@ var _ = Describe("Static", func() {
 
 	for _, trueOrFalse := range []bool{true, false} {
 		kubeIPVSEnabled := trueOrFalse
+		var dropAction Action
+		dropAction = DropAction{}
+		dropActionString := "DROP"
+		if kubeIPVSEnabled {
```
Comment (mgleung) [inline felix/rules/static_test.go:84]: nit: I get that this is functionally the same as using `trueOrFalse` but I think it would be a little more clear that this is not related to IPVS to have it use `trueOrFalse` instead of `kubeIPVSEnabled`

File: felix/rules/static_test.go [hunk 2]
```diff
@@ -78,6 +78,13 @@ var _ = Describe("Static", func() {
 
 	for _, trueOrFalse := range []bool{true, false} {
 		kubeIPVSEnabled := trueOrFalse
+		var dropAction Action
+		dropAction = DropAction{}
```
Comment (mgleung) [inline felix/rules/static_test.go:82]: nit: same as above for making changing variables with `drop` in them to `deny`.

File: libcalico-go/lib/validator/v3/validator_test.go [hunk 3]
```diff
@@ -679,6 +679,7 @@ func init() {
 		Entry("should accept a valid IptablesFilterAllowAction value 'Accept'", api.FelixConfigurationSpec{IptablesFilterAllowAction: "Accept"}, true),
 		Entry("should accept a valid IptablesMangleAllowAction value 'Return'", api.FelixConfigurationSpec{IptablesMangleAllowAction: "Return"}, true),
 		Entry("should reject an invalid IptablesMangleAllowAction value 'Drop'", api.FelixConfigurationSpec{IptablesMangleAllowAction: "Drop"}, false),
+		Entry("should accept a valid IptablesFilterDenyAction value 'Drop'", api.FelixConfigurationSpec{IptablesFilterDenyAction: "Drop"}, true),
```
Comment (mgleung) [inline libcalico-go/lib/validator/v3/validator_test.go:682]: we should probably add a test case here for testing if `Reject` is accepted too.

File: felix/config/config_params.go [hunk 4]
```diff
@@ -269,6 +269,7 @@ type Config struct {
 	DefaultEndpointToHostAction string `config:"oneof(DROP,RETURN,ACCEPT);DROP;non-zero,die-on-fail"`
 	IptablesFilterAllowAction   string `config:"oneof(ACCEPT,RETURN);ACCEPT;non-zero,die-on-fail"`
 	IptablesMangleAllowAction   string `config:"oneof(ACCEPT,RETURN);ACCEPT;non-zero,die-on-fail"`
+	IptablesFilterDenyAction    string `config:"oneof(DROP,REJECT);DROP;non-zero,die-on-fail,local"`
```
Comment (mgleung) [inline felix/config/config_params.go:272]: any specific reason we want to set `local` here?

**Review by mgleung — CHANGES_REQUESTED** [review]
Hey @olljanat , sorry for the delay. I just reviewed your PR but it'll likely need another rebase unfortunately.

**Comment by olljanat** [comment]
@mgleung thanks. Rebased and updated based on your feedback.

**Comment by olljanat** [comment]
@mgleung looks that I forgot re-generate manifests after rebase. Maybe now tests will really start?

**Comment by olljanat** [comment]
@mgleung Semaphore link here does not work for some reason but based on https://tigera.semaphoreci.com/branches/6a94d48a-8401-4717-9143-d8bd0843945d looks that CI is green 🟢

**Review by mgleung — APPROVED** [review]
@olljanat LGTM. Thanks for your patience and for working with us for all this time!
